plugins/flutter_aepedgeidentity/lib/src/aepedgeidentity_identity_map.dart

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import 'package:flutter_aepedgeidentity/src/aepedgeidentity_identity_item.dart';
+
+/// identity map containing a set of end user identities, keyed on either namespace integration code or the namespace ID of the identity.
+
+class IdentityMap {
+  ///add an `IdentityItem` to this `IdentityMap`
+  Map<String, List<IdentityItem>> identityMap = {};
+
+  void addItem(IdentityItem item, String namespace) {
+    if (item.id.isEmpty || item.id.length == 0) {
```

Comment:
id.isEmpty and id.length check the same thing. Do we need null check here?

Reply:
Item can't be null, so only checking isEmpty now.